File: VitrueWebApi/VitrueWebAPI/Models/Suggestion.cs
namespace VitrueWebAPI.Models
{
    public enum SuggestionType { Equipment, Exercise, Behavioural, Lifestyle }
    public enum SuggestionStatus { Pending, InProgress, Completed, Overdue }
    public enum SuggestionPriority { Low, Medium, High }
    public enum SuggestionSource { Vida, Admin }

    public record Suggestion
    {
        public required Guid Id { get; init; }
        public required Guid EmployeeId { get; init; }
        public required SuggestionType Type { get; init; }
        public required string Description { get; init; }
        public required SuggestionStatus Status { get; init; }
        public required SuggestionPriority Priority { get; init; }
        public required SuggestionSource Source { get; init; }
        public required DateTime DateCreated { get; init; }
        public required DateTime DateUpdated { get; init; }
        public string Notes { get; init; } = string.Empty;
    }
}

File: Frontend/src/types/suggestion.ts
export enum SuggestionType {
  Equipment = 'equipment',
  Exercise = 'exercise',
  Behavioural = 'behavioural',
  Lifestyle = 'lifestyle',
}

export enum SuggestionStatus {
  Pending = 'pending',
  InProgress = 'in_progress',
  Completed = 'completed',
  Overdue = 'overdue',
}

export enum SuggestionPriority {
  Low = 'low',
  Medium = 'medium',
  High = 'high',
}

export enum SuggestionSource {
  Vida = 'vida',
  Admin = 'admin',
}

export interface Suggestion {
  id: string
  employeeId: string
  type: SuggestionType
  description: string
  status: SuggestionStatus
  priority: SuggestionPriority
  source: SuggestionSource
  dateCreated: string
  dateUpdated: string
  notes: string
}

export interface SuggestionViewModel {
  id: string
  employeeName: string
  type: SuggestionType
  description: string
  status: SuggestionStatus
  priority: SuggestionPriority
  source: SuggestionSource
  dateCreated: string
  notes: string
}
